orchagent/portsorch.cpp

```diff
@@ -1239,14 +1241,71 @@ bool PortsOrch::setHostIntfsOperStatus(const Port& port, bool isUp) const
     return true;
 }
 
+void PortsOrch::updateDbPortFlapCounter(const Port& port, vector<FieldValueTuple>& old_tuples, vector<FieldValueTuple>& new_tuples) const
```

Comment:
store the counter in orchagent and update without querying the current value

Reply:
@stcheng Do you see any issues for warm-restart/warm-reboot case where the counters/ts in orchagent would be cleared even though from HW perspective things should stay the same before the warm-restart/reboot?

Reply:
@stcheng may be something to check upon before we make the suggested changes, as there might be some discrepancy then.

Reply:
@zhenggen-xu @prprakas we could query the counter number during the port initialization time to get the base number.

Reply:
read and set is better as user can then clear this counter, then it can get reset. however, i suggest to use redis incr command which is atomic.